File: charity_MGT_backend/models/charityModel.js
const mongoose = require('mongoose');
const validator = require("validator");

const charitySchema = new mongoose.Schema({
    name:{
        type:String,
        required:[true,'Please fill the name of charity']
    },
    description:{
        type:String,
        required:[true,'Please brief the description of charity']  
    },
    category:{
        type:mongoose.Schema.Types.ObjectId,
        ref:'Category',
        required:true
    },
    image:{
        type:String,
        default:"default.png"
    },
    email:{
        type:String,
        required:[true,'Please provide your charity email'] ,
        unique: true,
        lowercase: true,
        validate: [validator.isEmail, "Please provide a valid email"], 
    },
    address:{
        type:String,
        default:"adrress"
    },
    phone:{
        type:String,
        default:"phone"
    },
    NumOfDonors:{
        type:Number,
        default:0
    },
    SumofDonations:{
        type:Number,
        default:0
    }
    
},
    {
        timestamps:true
    }

)
Charity = mongoose.model('Charity',charitySchema);
module.exports = Charity;

File: charity_mobile_app/charity_management/lib/Data/Models/charity_model.dart
import 'package:equatable/equatable.dart';

class CharityModel extends Equatable{
  String id;
  String name;
  String description;
  String category;
  String image;
  String email;
  String address;
  String phone;
  int numOfDonors;
  int sumOfDonations;
  String createdAt;
  String updatedAt;

  CharityModel({this.id, this.name, this.description, this.category, this.image, this.email, this.address, this.phone, this.numOfDonors, this.sumOfDonations,this.createdAt,this.updatedAt});

  factory CharityModel.fromJson(Map<String, dynamic> parsedJson){
    return CharityModel(
      id: parsedJson['id'],
      name: parsedJson['name'],
      description: parsedJson['description'],
      category: parsedJson['category'],
      image: parsedJson['image'],
      email: parsedJson['email'],
      address: parsedJson['address'],
      phone: parsedJson['phone'],
      numOfDonors: parsedJson['NumOfDonors'],
      sumOfDonations: parsedJson['SumofDonations'],
      createdAt: parsedJson['createdAt'],
      updatedAt: parsedJson['updatedAt'],
    );
  }

  @override
  // TODO: implement props
  List<Object> get props => [id,name,description,category,image,email,address,phone,numOfDonors,sumOfDonations,createdAt,updatedAt];

}